smdebug/tensorflow/base_hook.py

```diff
@@ -420,7 +421,10 @@ def set_gradients(self, gradients=None, gradients_and_variables=None):
         # TF 2.x doesn't provide gradient/optimizer variable names and values by default.
         # Skipping set_gradients and set_optimizer_variables for Tf 2.x until there is
         # support to pass names and values from TF side.
-        if is_tf_version_2x() and tf.executing_eagerly():
+
+        # From TF 2.2, executing_eagerly_outside_functions() should be called using
+        # tf.compat.v1.executing_eagerly_outside_functions()
+        if is_tf_version_2x() and ops.executing_eagerly_outside_functions():
```

Comment:
Probably want to address the function name change now then - TF 2.2 will be released soon.

Reply:
this release will go into TF 2.1 container which wont have this name. so we'll have to make the change for the next container

Reply:
it fails now if i use tf.compat.v1

Reply:
smdebug should work for TF 2.1 and 2.2 simultaneously though, it's not just made for a specific container. Might need to wrap in a try/except or version check here.

Reply:
just verified with 2.2.0-rc0. there is no error when ops.executing_eagerly_outside_functions()
is used. 
in 2.2, both tf.compat.v1.executing_eagerly_outside_functions() and ops.executing_eagerly_outside_functions() will work.
in 2.1, only ops.executing_eagerly_outside_functions() will work. using tf.compat.v1.executing_eagerly_outside_functions() throws AttributeError exception

Reply:
will change the comment here